File: app/src/main/java/com/example/lightshop/HomeActivity.java
package com.example.lightshop;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import com.example.lightshop.utils.StatusBarUtils;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class HomeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        StatusBarUtils.applyWhiteStatusBar(this);
        setContentView(R.layout.activity_home);

        setupBottomNavigation();

        if (savedInstanceState == null) {
            String navigateTo = getIntent().getStringExtra("navigate_to");
            if ("orders".equals(navigateTo)) {
                switchToOrders();
            } else {
                loadFragment(new HomeFragment());
            }
        }
    }

    private void setupBottomNavigation() {
        BottomNavigationView bottomNav = findViewById(R.id.bottom_nav);
        bottomNav.setOnItemSelectedListener(item -> {
            int itemId = item.getItemId();
            Fragment selectedFragment = null;

            if (itemId == R.id.nav_home) {
                selectedFragment = new HomeFragment();
            } else if (itemId == R.id.nav_category) {
                selectedFragment = new CategoryFragment();
            } else if (itemId == R.id.nav_orders) {
                selectedFragment = new MyOrdersFragment();
            } else if (itemId == R.id.nav_profile) {
                selectedFragment = new ProfileFragment();
            }

            if (selectedFragment != null) {
                loadFragment(selectedFragment);
                return true;
            }
            return false;
        });
    }

    private void loadFragment(Fragment fragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragment_container, fragment)
                .commit();
    }

    public void openProductDetail(com.example.lightshop.models.ProductModel product) {
        android.content.Intent intent = new android.content.Intent(this, ProductDetailActivity.class);
        intent.putExtra("product", product);
        startActivity(intent);
    }

    public void switchToCategory() {
        BottomNavigationView bottomNav = findViewById(R.id.bottom_nav);
        bottomNav.setSelectedItemId(R.id.nav_category);
    }

    public void switchToOrders() {
        BottomNavigationView bottomNav = findViewById(R.id.bottom_nav);
        bottomNav.setSelectedItemId(R.id.nav_orders);
    }

    public void switchToProfile() {
        BottomNavigationView bottomNav = findViewById(R.id.bottom_nav);
        bottomNav.setSelectedItemId(R.id.nav_profile);
    }

    // --- Data Models (Keep here as fragments reference them) ---
    public static class HomeCategory {
        public String name;
        public int iconRes;
        public int bgRes;

        public HomeCategory(String name, int iconRes, int bgRes) {
            this.name = name;
            this.iconRes = iconRes;
            this.bgRes = bgRes;
        }
    }
}

File: app/src/main/java/com/example/lightshop/HomeActivity.kt
package com.example.lightshop

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class HomeActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)

        setupCategories()
        setupTopDeals()
        setupBottomNavigation()
    }

    private fun setupCategories() {
        val rvCategories = findViewById<RecyclerView>(R.id.rv_categories)
        val categories = listOf(
            Category("Men", R.drawable.ic_men, R.color.cat_men_bg),
            Category("Women", R.drawable.ic_women, R.color.cat_women_bg),
            Category("Electronics", R.drawable.ic_electronics, R.color.cat_electronics_bg),
            Category("Home", R.drawable.ic_home_cat, R.color.cat_home_bg),
            Category("Beauty", R.drawable.ic_beauty, R.color.cat_beauty_bg)
        )
        rvCategories.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        rvCategories.adapter = CategoryAdapter(categories)
    }

    private fun setupTopDeals() {
        val rvDeals = findViewById<RecyclerView>(R.id.rv_deals)
        val products = listOf(
            Product("Analog Watch", "₹599", "₹999", "-40%", R.drawable.ic_watch),
            Product("Sports Shoes", "₹1,299", "₹1,999", "-35%", R.drawable.ic_shoes),
            Product("Backpack", "₹749", "₹999", "-25%", R.drawable.ic_backpack)
        )
        rvDeals.layoutManager = LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, false)
        rvDeals.adapter = ProductAdapter(products)
    }

    private fun setupBottomNavigation() {
        // Implementation for navigation item selection can be added here
    }

    // --- Data Models ---
    data class Category(val name: String, val iconRes: Int, val bgRes: Int)
    data class Product(val name: String, val price: String, val oldPrice: String, val discount: String, val imageRes: Int)

    // --- Adapters ---
    inner class CategoryAdapter(private val items: List<Category>) : RecyclerView.Adapter<CategoryAdapter.ViewHolder>() {
        inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
            val name: TextView = view.findViewById(R.id.cat_name)
            val icon: ImageView = view.findViewById(R.id.cat_icon)
            val bg: View = view.findViewById(R.id.cat_bg)
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.item_category, parent, false)
            return ViewHolder(view)
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            val item = items[position]
            holder.name.text = item.name
            holder.icon.setImageResource(item.iconRes)
            holder.bg.background.setTint(holder.itemView.context.getColor(item.bgRes))
        }

        override fun getItemCount() = items.size
    }

    inner class ProductAdapter(private val items: List<Product>) : RecyclerView.Adapter<ProductAdapter.ViewHolder>() {
        inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
            val name: TextView = view.findViewById(R.id.tv_product_name)
            val price: TextView = view.findViewById(R.id.tv_price)
            val oldPrice: TextView = view.findViewById(R.id.tv_old_price)
            val discount: TextView = view.findViewById(R.id.tv_discount)
            val image: ImageView = view.findViewById(R.id.iv_product)
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.item_product, parent, false)
            return ViewHolder(view)
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            val item = items[position]
            holder.name.text = item.name
            holder.price.text = item.price
            holder.oldPrice.text = item.oldPrice
            holder.discount.text = item.discount
            holder.image.setImageResource(item.imageRes)
        }

        override fun getItemCount() = items.size
    }
}